Makefile

```diff
@@ -11,7 +11,8 @@ cabalver = 3.12.1.0
 # The LLVM version GHC requires depends on GHC's version,
 # so make sure adjust the version when you tweak GHC version!
 ghcup_apt_dependency = build-essential curl libffi-dev libffi8ubuntu1 \
-	libgmp-dev libgmp10 libncurses-dev libncurses6 libtinfo6 llvm-15
+	libgmp-dev libgmp10 libncurses-dev libncurses6 libtinfo6 llvm-15 \
+	libgsl0-dev liblapack-dev libatlas-base-dev libglpk-dev
```

Comment:
This line's bugging me for maintainability and rationale. I know that apt-installation should be done in one place for readability, but ghcup Dockerfile and serverproto Dockerfile are separated to offer toolgen a vanilla workspace (a weak claim that apt-installation should be moved to serverproto).

For maintainability, whether or not the apt-installation for `hmatrix` is moved to serverproto Dockerfile after the GHCup installation, there should be another variable. We need to make clear that the reason for the packages is not GHCup but hmatrix.

Reply:
The reason why I'm saying my first suggestion is weak is because we need to place apt-installation after the installation of GHCup in installation script to make Docker image exactly immitate the installation. Docker images are to immitate the installation, but this is making things other way round; We're bending the installation to align it with the immitation.

Reply:
I guess it's OK to run `apt-get` twice on the AtCoder server, as long as the installation script is kept clear. I'll try moving the `hmatrix` installation from ghcup `Dockerfile` to the serverproto `Dockerfile` with my zero Docker skill.